src/main/java/com/minecolonies/coremod/util/BlockPosUtil.java

```diff
@@ -127,6 +129,58 @@ public static BlockPos readFromByteBuf(@NotNull final ByteBuf buf)
     }
 
     /**
+     * this checks that you are not in liquid.  Will check for all liquids, even those from other mods
+     * before TP
+     *
+     * @param blockPos for the current block LOC
+     * @param sender uses the player to get the world
+     * @return isSafe true=safe false=water or lava
+     */
+    public static boolean isPositionSafe(@NotNull ICommandSender sender, BlockPos blockPos)
+    {
+        return sender.getEntityWorld().getBlockState(blockPos).getBlock() != Blocks.AIR
+                && !sender.getEntityWorld().getBlockState(blockPos).getMaterial().isLiquid()
+                && !sender.getEntityWorld().getBlockState(blockPos.up()).getMaterial().isLiquid();
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)